extensions/spark/kyuubi-spark-connector-hive/src/test/scala/org/apache/kyuubi/spark/connector/hive/command/DropNamespaceSuite.scala

```diff
@@ -70,10 +71,15 @@ trait DropNamespaceSuiteBase extends DDLCommandTestUtils {
     checkNamespace(Seq(namespace) ++ builtinNamespace)
 
     // $catalog.ns.table is present, thus $catalog.ns cannot be dropped.
-    val e = intercept[IllegalStateException] {
+    val e = intercept[AnalysisException] {
       sql(s"DROP NAMESPACE $catalogName.$namespace")
     }
-    assert(e.getMessage.contains(s"Namespace $namespace is not empty"))
+    if (SPARK_RUNTIME_VERSION >= "3.4") {
+      assert(e.getMessage().contains(
+        s"[SCHEMA_NOT_EMPTY] Cannot drop a schema `$namespace` because it contains objects"))
+    } else {
+      assert(e.getMessage().contains("Use CASCADE option to drop a non-empty database"))
+    }
```

Comment:
```
    interceptContains[AnalysisException] {
      sql(s"DROP NAMESPACE $catalogName.$namespace")
    }(if (SPARK_RUNTIME_VERSION >= "3.4") {
      s"[SCHEMA_NOT_EMPTY] Cannot drop a schema `$namespace` because it contains objects"
    } else {
      "Use CASCADE option to drop a non-empty database"
    })
```

Consider the new-added helper method `interceptContains` available in AssertionUtils for asserting error message.

Reply:
Oh, thanks~

Reply:
😀

Reply:
LGTM. I think it's ready to merge.

Reply:
Thanks bowen, waitting CI